src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -744,7 +755,17 @@ impl Step for Clippy {
 
     /// Runs `cargo test` for clippy.
     fn run(self, builder: &Builder<'_>) {
-        let stage = self.stage;
+        // If stage is explicitly set or not lower than 2, keep it. Otherwise, make sure it's at least 2
+        // as tests for this step don't work with a lower stage.
+        let stage = if builder.config.explicit_stage_from_cli
```

Comment:
Maybe we could add a method like `has_explicit_stage()`, to avoid having to remember to check both the options?